x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import type { JobSavedObjectService } from './service';
+import { ML_SAVED_OBJECT_TYPE } from './saved_objects';
+import { JobType } from '../../common/types/saved_objects';
+
+import { Job } from '../../common/types/anomaly_detection_jobs';
+import { Datafeed } from '../../common/types/anomaly_detection_jobs';
+import { DataFrameAnalyticsConfig } from '../../common/types/data_frame_analytics';
+
+interface JobSavedObjectStatus {
+  jobId: string;
+  type: JobType;
+  datafeedId?: string | null;
+  namespaces: string[] | undefined;
+  checks: {
+    jobExists: boolean;
+    datafeedExists?: boolean;
+  };
+}
+
+interface JobStatus {
+  jobId: string;
+  datafeedId?: string | null;
+  checks: {
+    savedObjectExits: boolean;
+  };
+}
+
+interface SavedObjectJob {
+  [ML_SAVED_OBJECT_TYPE]: {
+    job_id: string;
+    type: JobType;
+  };
+  namespaces: string[];
+}
+
+interface StatusResponse {
+  savedObjects: {
+    [type in JobType]: JobSavedObjectStatus[];
+  };
+  jobs: {
+    [type in JobType]: JobStatus[];
+  };
+}
+
+export function checksFactory(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+) {
+  async function checkStatus(): Promise<StatusResponse> {
+    const jobObjects = await jobSavedObjectService.getAllJobObjects(undefined, false);
+    // load all non-space jobs and datafeeds
+    const { body: adJobs } = await client.asInternalUser.ml.getJobs<{ jobs: Job[] }>();
+    const { body: datafeeds } = await client.asInternalUser.ml.getDatafeeds<{
+      datafeeds: Datafeed[];
+    }>();
+    const { body: dfaJobs } = await client.asInternalUser.ml.getDataFrameAnalytics<{
+      data_frame_analytics: DataFrameAnalyticsConfig[];
+    }>();
+
+    const savedObjectsStatus: JobSavedObjectStatus[] = jobObjects.map(
+      ({ attributes, namespaces }) => {
+        const type: JobType = attributes.type;
+        const jobId = attributes.job_id;
+        const datafeedId = type === 'anomaly-detector' ? attributes.datafeed_id : undefined;
```

Comment:
Suggestion to move types like 'anomaly-detector` to constants since they are used in multiple places.